specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/privateEndpoint.json

```diff
@@ -567,6 +789,115 @@
+          "type": "string",
+          "description": "Resource type."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PrivateDNSZoneGroupProperties",
+          "description": "Properties of the private dns zone group."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Private dns zone group resource."
+    },
+    "PrivateDNSZoneGroupProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the private dns zone group resource."
+        },
+        "privateDnsZoneConfigs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateDnsZoneConfig"
+          },
+          "description": "A collection of private dns zone configurations of the private dns zone group."
+        }
+      },
+      "description": "Properties of the private dns zone group."
+    },
+    "PrivateDnsZoneConfig": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PrivateDnsZoneConfigProperties",
+          "description": "Properties of the private dns zone configuration."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "PrivateDnsZoneConfig resource."
+    },
+    "PrivateDnsZoneConfigProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the private dns zone group resource."
+        },
+        "privateDnsZoneId": {
+          "type": "string",
+          "description": "The resource id of the private dns zone."
+        }, 
+        "recordSets": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/recordSet"
+          },
+          "description": "A collection of information regarding a recordSet, holding information to identify private resources"
+        }
+      }
+    },
+    "recordSet": {
+      "properties": {
+        "recordType": {
+          "type": "string",
+          "description": "resource record type"
+        },
+        "recordSetName": { 
+          "type": "string",
+          "description": "recordset name"
+        },
+        "fqdn": { 
+          "type": "string",
```

Comment:
should it be constrained to these?https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-overview#dns-configuration

Reply:
Sorry I'm not sure what this means, are you maybe saying the fqdn has to be constrained to the host, privzone, custom dns zone options? I'm not sure what the would mean ultimately for the code.

Reply:
since it's the private dns for private endpoint, I think the descriptions of `fqdn` and `ipAddresses` should be same with those descriptions in `CustomDnsConfig`

Reply:
I guess I'm confused.. Looking at the documentation given to me by my manager: ([https://microsoft.sharepoint.com/:w:/r/teams/WAG/AzureNetworking/_layouts/15/Doc.aspx?sourcedoc=%7B3C240246-0538-4653-9EBA-6F640E7BA5AE%7D&file=Private%20Endpoint%20DNS%20Zone%20Integration.docx&action=default&mobileredirect=true]) , it says "Customer creates a private zone (on DNS RP) with name derived from the FQDN suffix of the service  "

But in your documentation it states "You can use private DNS zones to override the DNS resolution for a given private endpoint. A private DNS zone can be linked to your virtual network to resolve specific domains." 

I'm following up with my manager about this to confirm if this is for the privateendpoint or for the private link resources.

Reply:
Got it, I've updated the definitions, thank you so much for your feedback.